probe-rs/src/architecture/arm/core/registers/aarch32.rs

```diff
@@ -36,212 +36,215 @@ pub(crate) static AARCH32_WITH_FP_32_CORE_REGSISTERS: Lazy<CoreRegisters> = Lazy
         id: RegisterId(38),
         data_type: RegisterDataType::FloatingPoint(64),
+        unwind_rule: UnwindRule::Clear,
     },
     CoreRegister {
-        name: "D22",
-        roles: &[RegisterRole::FloatingPoint],
+        roles: &[RegisterRole::Core("D22"), RegisterRole::FloatingPoint],
         id: RegisterId(39),
         data_type: RegisterDataType::FloatingPoint(64),
+        unwind_rule: UnwindRule::Clear,
     },
     CoreRegister {
-        name: "D23",
-        roles: &[RegisterRole::FloatingPoint],
+        roles: &[RegisterRole::Core("D23"), RegisterRole::FloatingPoint],
         id: RegisterId(40),
         data_type: RegisterDataType::FloatingPoint(64),
+        unwind_rule: UnwindRule::Clear,
     },
     CoreRegister {
-        name: "D24",
-        roles: &[RegisterRole::FloatingPoint],
+        roles: &[RegisterRole::Core("D24"), RegisterRole::FloatingPoint],
         id: RegisterId(41),
         data_type: RegisterDataType::FloatingPoint(64),
+        unwind_rule: UnwindRule::Clear,
     },
     CoreRegister {
-        name: "D25",
-        roles: &[RegisterRole::FloatingPoint],
+        roles: &[RegisterRole::Core("D25"), RegisterRole::FloatingPoint],
         id: RegisterId(42),
         data_type: RegisterDataType::FloatingPoint(64),
+        unwind_rule: UnwindRule::Clear,
     },
     CoreRegister {
-        name: "D26",
-        roles: &[RegisterRole::FloatingPoint],
+        roles: &[RegisterRole::Core("D26"), RegisterRole::FloatingPoint],
         id: RegisterId(43),
         data_type: RegisterDataType::FloatingPoint(64),
+        unwind_rule: UnwindRule::Clear,
     },
     CoreRegister {
-        name: "D27",
-        roles: &[RegisterRole::FloatingPoint],
+        roles: &[RegisterRole::Core("D27"), RegisterRole::FloatingPoint],
         id: RegisterId(44),
         data_type: RegisterDataType::FloatingPoint(64),
+        unwind_rule: UnwindRule::Clear,
     },
     CoreRegister {
-        name: "D28",
-        roles: &[RegisterRole::FloatingPoint],
+        roles: &[RegisterRole::Core("D28"), RegisterRole::FloatingPoint],
         id: RegisterId(45),
         data_type: RegisterDataType::FloatingPoint(64),
+        unwind_rule: UnwindRule::Clear,
     },
     CoreRegister {
-        name: "D29",
-        roles: &[RegisterRole::FloatingPoint],
+        roles: &[RegisterRole::Core("D29"), RegisterRole::FloatingPoint],
         id: RegisterId(46),
         data_type: RegisterDataType::FloatingPoint(64),
+        unwind_rule: UnwindRule::Clear,
     },
     CoreRegister {
-        name: "D30",
-        roles: &[RegisterRole::FloatingPoint],
+        roles: &[RegisterRole::Core("D30"), RegisterRole::FloatingPoint],
         id: RegisterId(47),
         data_type: RegisterDataType::FloatingPoint(64),
+        unwind_rule: UnwindRule::Clear,
     },
     CoreRegister {
-        name: "D31",
-        roles: &[RegisterRole::FloatingPoint],
+        roles: &[RegisterRole::Core("D31"), RegisterRole::FloatingPoint],
```

Comment:
Can we make those names an enum too?

Reply:
Honestly, I think there are too many variations accross the architectures. They used to be strings when it was just the "name" field, but in a previous PR Tiwalun agreed that it would make sense to move into RegisterRole, to simplify handling. See: https://github.com/probe-rs/probe-rs/pull/1627#discussion_r1222448359

Reply:
So I agree that there is a lot of variation. But it's nice to have an enum to give the user a list of possible values. Especially for things like SP, PC, etc. But I see that we would need many variants for the different architectures, which then complicate things. We can think of it later and leave it be for now.